File: Python x EDAII/Ordenamiento/ShellSort.py
class ShellSort:

	arr = []

	def __init__(self, arr):
		'Constructor de la clase'
		self.arr = arr

	def shellSort(self):
		'Algoritmo de ordenamiento'
		n = len(self.arr)
		h = n//2

		# Genera las sublistas correspondientes y a partir del elemento medio realiza Insertion Sort
		while h > 0:
			# Insertion Sort
			for i in range(h,n):
				tmp = self.arr[i]
				j = i
				# Si la parte izquierda a analizar ya está ordenada, sale del ciclo
				while j>=h and self.arr[j-h]>tmp:
					self.arr[j] = self.arr[j-h]
					j -= h
				self.arr[j] = tmp	 # Coloca el elemento inicial en su posición correcta
			h //= 2

# Ejecutando el objeto
programa = ShellSort([13,8,21,6,10,2])
programa.shellSort()
print(programa.arr)

File: EDAII x Java/Ordenamiento/ShellSort.java
public class ShellSort{

	int[] arr = {13,8,21,6,10,2};

	public static void main(String[] args){
		ShellSort ss = new ShellSort();
		ss.shellSort(ss.arr); // Ordenamiento ShellSort

		for (int i=0; i<ss.arr.length; i++){
			System.out.println(ss.arr[i]);
		}
	}

	public void shellSort(int[] A){
		int i, j, h, tmp, n = A.length;

		// Genera las sublistas correspondientes y a partir del elemento medio realiza Insertion Sort
		for (h=n/2; h>0; h=h/2){
			// Insertion Sort
			for (i=h; i<n; i++){
				tmp = A[i];
				// Si la parte izquierda a analizar ya está ordenada, sale del ciclo
				for (j=i; j>=h && A[j-h]>tmp; j=j-h){
					A[j] = A[j-h];
				}
				A[j] = tmp; // Coloca el elemento inicial en su posición correcta
			}
		}
	}

}